### PR TITLE
Moved method definition to interface

### PR DESCRIPTION
**Pull request by core23**
This will be a bc, but I think it's better to have all public methods in the interface.

### CODE REVIEW

**Comment by OskarStark** [comment]
Only if they are really needed, IMO the interface should just define the minimal amount of methods you need.

**Comment by core23** [comment]
It will be more common to ``remove`` something then setting something completly to null. 

When they shouldn't be accessed outside, the methods could be private or protected.

**Comment by OskarStark** [comment]
the could, but don't have to, to implement the interface

what do you say @Soullivaneuh

**Comment by Soullivaneuh** [comment]
I don't know enough this bundle to say if these methods should be on interface or not.

In any way yes, this is BC break on would be merged only for major.

Any chance to have BC handling @core23?

**Comment by core23** [comment]
No idea how to handle this break. You could introduce a new ``AdvancedSeoPageInterface`` but this would get messed by time...

I recently read the [symfony docs](http://symfony.com/doc/current/contributing/code/bc.html) for bc breaks. They handle this changes with care, but you were allowed to add methods to interfaces when adding a hint to the **UPGRADE** file. 

What I got from your approach would mean, that changes on interfaces are only allowed any few years when a new mayor version gets released.

**Comment by Soullivaneuh** [comment]
@core23 Please rebase.

**Comment by OskarStark** [comment]
@Soullivaneuh i think we can merge this in 3.x, right?

**Comment by Soullivaneuh** [comment]
> @Soullivaneuh i think we can merge this in 3.x, right?

Yes, `master` **is** `3.x`.

**Comment by OskarStark** [comment]
Sorry i was confused across all the bundles :neckbeard::grimacing:

**Comment by Soullivaneuh** [comment]
@core23 Could you please rebase and fill the needs labels?

**Comment by core23** [comment]
Done @Soullivaneuh

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,3 +4,6 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 ### Added
+
+### Changed
+- moved public methods in `SeoPage` to `SeoPageInterface`
```
Comment (Soullivaneuh) [inline CHANGELOG.md]: **M**oved

**Comment by Soullivaneuh** [comment]
@core23 Your upgrade not must be on `UPGRADE-4.0.md`, not 3.0.

**Comment by core23** [comment]
fixed

**Comment by Soullivaneuh** [comment]
RTM. :+1: 

Feel free to merge it once CI will be green.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,3 +4,6 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 ### Added
+
+### Changed
+- Moved public methods in `SeoPage` to `SeoPageInterface`
```
Comment (Soullivaneuh) [inline CHANGELOG.md]: Just one thing again:

`Moved public methods in`

Should not be **from** instead of in?

**Pull request merged**